### PR TITLE
added optional voice typing button

### PR DESCRIPTION
**Pull request by ronniedroid**
Added the option to enable voice input.
I recommend using this with  the FUTU Voice Input IME.

### CODE REVIEW

File: app/src/main/kotlin/org/fossify/keyboard/services/SimpleKeyboardIME.kt [hunk 1]
```diff
@@ -488,4 +493,21 @@ class SimpleKeyboardIME : InputMethodService(), OnKeyboardActionListener, Shared
 
         return Icon.createWithData(byteArray, 0, byteArray.size)
     }
+
+    // voice input
+
+    private fun getVoiceTypingIm(imm: InputMethodManager): AbstractMap.SimpleEntry<String, InputMethodSubtype>? {
+        val enabledKeyboards = imm.enabledInputMethodList
+        for (im in enabledKeyboards) for (imst in imm.getEnabledInputMethodSubtypeList(im, true))
+            if (imst.mode == "voice") return AbstractMap.SimpleEntry<String, InputMethodSubtype>(im.id, imst)
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/keyboard/services/SimpleKeyboardIME.kt:502]: Please don't do nested loops on one line, they are hard to read. Also, always use braces even if it's a one-liner.

**Comment by naveensingh** [comment]
There are some other things like the voice IME is only launched on Pie plus but the preference is shown regardless of Android version so I'm working on it myself now as I want to release the keyboard soon. I'll push the changes here and then merge, thanks!

**Comment by ronniedroid** [comment]
Ah, sorry I missed that, I am in the process of moving to a new city and don't have much time, so thanks for doing it yourself.

**Comment by naveensingh** [comment]
No problem. I'm about to move too, a friend of mine once said moving is like a black hole, it sucks all of your spacetime.

**Comment by naveensingh** [comment]
Replaced the toggle preference with a radio group for more flexibility:

image

**Pull request merged**